src/modules/land_detector/MulticopterLandDetector.cpp

```diff
@@ -154,40 +149,88 @@ bool MulticopterLandDetector::_get_ground_contact_state()
 		return true;
 	}
 
+	const bool lpos_available = (hrt_elapsed_time(&_vehicle_local_position.timestamp) < 1_s);
+
 	// land speed threshold
-	float land_speed_threshold = 0.9f * math::max(_params.landSpeed, 0.1f);
+	const float land_speed_threshold = 0.9f * math::max(_params.landSpeed, 0.1f);
+
 
-	// Check if we are moving vertically - this might see a spike after arming due to
-	// throttle-up vibration. If accelerating fast the throttle thresholds will still give
-	// an accurate in-air indication.
 	bool vertical_movement = false;
```

Comment:
I don't think that if there is no local position, assuming that there is no vertical movement as default is good default.
(there is a similar comment about horizontal movement bellow)

Reply:
In master now one of the conditions for ground_contact is `!vertical_movement || !_has_altitude_lock()`, so it's equivalent. Perhaps more importantly, the `ground_contact` state isn't even used by the position controller or `_get_maybe_landed_state()` if there's no local vertical position.

So with that in mind I'm not sure which is less bad.

The more I dig into this the more I want to take a clean pass and untangle all this logic. I'm trying to remember we're pretty close to resolving the immediate bug(s) on hand.

Reply:
@dagar I am not sure about what you answered, should we assume that there is no vertical movment as default, or not?
how the logic later (on position controller) treat threat data, is another thing.

Reply:
Defaulting to these to `true` for now is also fine (makes no difference currently).